src/integrationTest/java/org/opensearch/security/http/AsyncTests.java

```diff
@@ -0,0 +1,115 @@
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.opensearch.test.framework.TestSecurityConfig.AuthcDomain.AUTHC_HTTPBASIC_INTERNAL;
+import static org.opensearch.test.framework.TestSecurityConfig.Role.ALL_ACCESS;
+import static org.opensearch.test.framework.cluster.TestRestClientConfiguration.getBasicAuthHeader;
+
+@RunWith(com.carrotsearch.randomizedtesting.RandomizedRunner.class)
+@ThreadLeakScope(ThreadLeakScope.Scope.NONE)
+public class AsyncTests {
+    private static final TestSecurityConfig.User ADMIN_USER = new TestSecurityConfig.User("admin").backendRoles("admin");
+
+    @ClassRule
+    public static LocalCluster cluster = new LocalCluster.Builder().singleNode()
+        .authc(AUTHC_HTTPBASIC_INTERNAL)
+        .users(ADMIN_USER)
+        .rolesMapping(new RolesMapping(ALL_ACCESS).backendRoles("admin"))
+        .anonymousAuth(false)
+        .nodeSettings(Map.of(ConfigConstants.SECURITY_RESTAPI_ROLES_ENABLED, List.of(ALL_ACCESS.getName())))
+        .build();
+
+    @Test
+    public void testBulkAndCacheInvalidationMixed() throws Exception {
+        String indexName = "test-index";
+        final String invalidateCachePath = "/_plugins/_security/api/cache";
+        final String nodesPath = "/_nodes";
+        final String bulkPath = "/_bulk";
+        final String document = ("{ \"index\": { \"_index\": \"" + indexName + "\" }}\n{ \"foo\": \"bar\" }\n").repeat(5);
+        final int parallelism = 5;
+        final int totalNumberOfRequests = 30;
+
+        try (final TestRestClient client = cluster.getRestClient()) {
+            IndexOperationsHelper.createIndex(cluster, indexName);
+
+            final CountDownLatch countDownLatch = new CountDownLatch(1);
+
+            final var cacheInvalidationRequests = AsyncActions.generate(() -> {
+                countDownLatch.await();
+                System.err.println("Generation triggered invalidate cache requests");
+                final HttpDelete delete = new HttpDelete(client.getHttpServerUri() + invalidateCachePath);
+                return client.executeRequest(delete, getBasicAuthHeader(ADMIN_USER.getName(), ADMIN_USER.getPassword()));
+            }, parallelism, totalNumberOfRequests);
+
+            final var bulkRequests = AsyncActions.generate(() -> {
+                countDownLatch.await();
+                System.err.println("Generation triggered bulk requests");
+                final HttpPost post = new HttpPost(client.getHttpServerUri() + bulkPath);
+                post.setEntity(new ByteArrayEntity(document.getBytes(StandardCharset.UTF_8), ContentType.APPLICATION_JSON));
+                return client.executeRequest(post, getBasicAuthHeader(ADMIN_USER.getName(), ADMIN_USER.getPassword()));
+            }, parallelism, totalNumberOfRequests);
+
+            final var nodesRequests = AsyncActions.generate(() -> {
+                countDownLatch.await();
+                System.err.println("Generation triggered node requests");
```

Comment:
Nit: I think I set a bad precedence when I didn't remove this in the other test case, I'd remove

Reply:
removed